libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1689,7 +1689,12 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 				functionType.kind() == FunctionType::Kind::DelegateCall
 			)
 				define(IRVariable{_memberAccess}, IRVariable(_memberAccess.expression()).part("functionSelector"));
-			else if (functionType.kind() == FunctionType::Kind::Declaration)
+			else if (
+				functionType.kind() == FunctionType::Kind::Declaration ||
+				// In some situations, internal function types also provide the "selector" member.
+				// See Types.cpp for details.
+				functionType.kind() == FunctionType::Kind::Internal
```

Comment:
Is this a blanket provision for all internal functions? Since the title of the PR is "...for some internal functions", I wonder how and where the "some" are filtered?

Reply:
They are filtered in the type checker. I can add some assertions to be sure it is not extended. On the other Hand, if there is no selector, then this will throw anyway.

Reply:
Okay. I just took a look at these lines https://github.com/ethereum/solidity/blob/eacf7c1cf98ea85c7acb97edd29c0f062940b96a/libsolidity/ast/Types.cpp#L3151-L3163

So we permit selector for internal functions if function is part of external interface and invoked outside the scope of the contract it is defined in. Could we assert in this PR that the function type actually belongs to external interface of the contract or is that information already lost in IR? I suppose scope information is lost?

Reply:
Added the assertion about the external interface.